File: Java/SortArrayByParity_905.java
import java.util.Arrays;

class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int l=0,r=nums.length-1;
        while(l<r){
            if(nums[l]%2==0){
                l++;
            }else{
                int temp=nums[l];
                nums[l]=nums[r];
                nums[r]=temp;
                r--;
            }
        }
        return nums;
    }
}

public class SortArrayByParity_905{
    public static void main(String[] args) {
        Solution obj=new Solution();

        int[] nums={3,1,2,4};
        int[] result = obj.sortArrayByParity(nums);
        System.out.println("Array satisfies the given condition: " + Arrays.toString(result));
    }
}

File: LeetCode/Python/SortArrayByParity_905.py

'''
905. Sort Array By Parity

'''

from typing import List

class Solution:
    def sortArrayByParity(self,nums:List[int]) -> List[int]:
        l=0
        r=len(nums)-1
        while l<r:
            if nums[l]%2==0:
                l+=1
            else:
                nums[l],nums[r]=nums[r],nums[l]
                r-=1
        return nums
    
s=Solution()

nums=[3,1,2,4]

print("Array satisfies the given condition :",s.sortArrayByParity(nums))

# Input: nums = [3,1,2,4]
# Output: [2,4,3,1]

# Input: nums = [0]
# Output: [0]